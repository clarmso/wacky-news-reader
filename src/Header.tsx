import React from "react";

import { Typography } from "@material-ui/core";

export type HeaderProps = {
  date: string;
};

const Header: React.FC<HeaderProps> = ({ date }) => {
  return (
    <>
      <Typography variant="h2" component="h1" align="center">
        🗞️ Wacky News Reader 🗞️
      </Typography>
      <Typography component="h6" align="center">
        Last Updated: {date}
      </Typography>
    </>
  );
};

export default Header;
