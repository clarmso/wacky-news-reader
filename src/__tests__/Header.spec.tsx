import { Provider } from "react-redux";
import { createStore } from "redux";
import { render } from "@testing-library/react";
import Header from "../Header";
import wackyReducer, { initialState } from "../connect/reducer";

it("renders with initial state", () => {
  const store = createStore(wackyReducer, initialState);
  const { asFragment } = render(
    <Provider store={store}>
      <Header />
    </Provider>
  );
  const renderedHtml = asFragment();
  expect(renderedHtml).toMatchSnapshot();
});

it("renders with 90s mode", () => {
  const store = createStore(wackyReducer, { ...initialState, is90s: true });
  const { asFragment } = render(
    <Provider store={store}>
      <Header />
    </Provider>
  );
  const renderedHtml = asFragment();
  expect(renderedHtml).toMatchSnapshot();
});
