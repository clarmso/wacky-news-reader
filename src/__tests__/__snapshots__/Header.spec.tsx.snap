// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with 90s mode 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        data-cy="about-link"
      >
        About
      </a>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <img
        alt="Frog reading news"
        src="newspaperfrog.gif"
      />
      <img
        alt="Wacky News Reader Header"
        src="wordart.png"
      />
      <img
        alt="Flipping newspaper"
        src="newspaper.gif"
      />
      <h6
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders with initial state 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        data-cy="about-link"
      >
        About
      </a>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h1
        class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter"
      >
        🗞️ Wacky News Reader 🗞️
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      />
    </div>
  </div>
</DocumentFragment>
`;
